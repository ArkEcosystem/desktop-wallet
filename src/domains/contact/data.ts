export const contact1 = {
	name: () => "ROBank",
	addresses: () => [
		{
			coin: "Ark",
			network: "mainnet",
			address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
			avatar: "avatar",
			isDelegate: () => true,
			isBusiness: () => true,
			isBridgechain: () => true,
		},
	],
};

export const contact2 = {
	name: () => "OLEBank",
	addresses: () => [
		{
			coin: "Ark",
			network: "mainnet",
			address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
			avatar: "avatar",
			isDelegate: () => true,
			isBusiness: () => false,
			isBridgechain: () => false,
		},
		{
			coin: "Bitcoin",
			network: "mainnet",
			address: "15pyr1HRAxpq3x64duXav1csmyCtXXu9G8",
			avatar: "avatar",
			isDelegate: () => false,
			isBusiness: () => false,
			isBridgechain: () => false,
		},
		{
			coin: "Ethereum",
			network: "mainnet",
			address: "0x5e8f7a63e31c759ef0ad5e71594e838b380d7c33",
			avatar: "avatar",
			isDelegate: () => false,
			isBusiness: () => true,
			isBridgechain: () => false,
		},
	],
};

export const contacts = [contact1, contact2];

export const networks = [
	{
		icon: "Ark",
		name: "ARK Ecosystem",
		className: "text-theme-danger-400 border-theme-danger-200",
	},
	{
		icon: "Bitcoin",
		name: "Bitcoin",
		className: "text-theme-warning-400 border-theme-warning-200",
	},
	{
		icon: "Ethereum",
		name: "Ethereum",
		className: "text-theme-neutral-800 border-theme-neutral-600",
	},
	{
		icon: "Lisk",
		name: "Lisk",
		className: "text-theme-primary-600 border-theme-primary-400",
	},
	{
		icon: "Ripple",
		name: "Ripple",
		className: "text-theme-primary-700 border-theme-primary-500",
	},
];
