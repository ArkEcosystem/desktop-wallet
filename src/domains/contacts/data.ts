export const contact1 = {
  name: () => "ROBank",
  addresses: () => [
    {
      coin: "Ark",
      network: "mainnet",
      address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
      isDelegate: () => true,
      isBusiness: () => true,
      isBridgechain: () => true,
    },
  ],
};

export const contact2 = {
  name: () => "OLEBank",
  addresses: () => [
    {
      coin: "Ark",
      network: "mainnet",
      address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
      isDelegate: () => true,
      isBusiness: () => false,
      isBridgechain: () => false,
    },
    {
      coin: "Bitcoin",
      network: "mainnet",
      address: "15pyr1HRAxpq3x64duXav1csmyCtXXu9G8",
      isDelegate: () => false,
      isBusiness: () => false,
      isBridgechain: () => false,
    },
    {
      coin: "Ethereum",
      network: "mainnet",
      address: "0x5e8f7a63e31c759ef0ad5e71594e838b380d7c33",
      isDelegate: () => false,
      isBusiness: () => true,
      isBridgechain: () => false,
    },
  ],
};

export const contacts = [
  contact1,
  contact2,
];
