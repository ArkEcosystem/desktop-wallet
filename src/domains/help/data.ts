export const categories = [
	{
		title: "Profile",
		subtitle: "38 topics",
		icon: "Profile",
		path: "/",
	},
	{
		title: "Portfolio",
		subtitle: "9 topics",
		icon: "Portfolio",
		path: "/",
	},
	{
		title: "Plugins",
		subtitle: "106 topics",
		icon: "Plugins",
		path: "/",
	},
	{
		title: "Exchanges",
		subtitle: "28 topics",
		icon: "Exchange",
		path: "/",
	},
];

export const searchData = [
	{
		title: "How do I install the plugin?",
		category: "Plugin",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as the bla bla bla",
	},
	{
		title: "Installing the side plugin",
		category: "Plugin",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as the bla bla bla",
	},
	{
		title: "If the plugin is not installed, what do we do?",
		category: "Plugin",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as the bla bla bla",
	},
];

export const helpfulArticles = [
	{
		title: "How to register an account?",
		path: "/",
	},
	{
		title: "How to enter your account",
		path: "/",
	},
	{
		title:
			"How can I change my personal data if I do not have access to my account (forgot my password) and to my mail?",
		path: "/",
	},
	{
		title:
			"If you need to make changes to the data entered during the passage of verification, where to apply, how to issue a request?",
		path: "/",
	},
	{
		title: "Can I delete my account",
		path: "/",
	},
];

export const popularArticles = [
	{
		title: "How to delete YubiKey for Two-factor Authentication (2FA)",
		path: "/",
	},
	{
		title: "Using a YubiKey for Two-Factor Authentication (2FA)",
		path: "/",
	},
	{
		title: "2FA Code Error",
		path: "/",
	},
	{
		title: "How to Reset SMS Authentication",
		path: "/",
	},
	{
		title: "Supported SMS countries Not Receiving SMS Auth Codes",
		path: "/",
	},
];

export const newestArticles = [
	{
		title: "How to withdraw from wallet",
		path: "/",
	},
	{
		title: "How to Make Internal Transfer within wallet",
		path: "/",
	},
	{
		title: "Missing Withdrawals",
		path: "/",
	},
	{
		title: "Withdrawal to a Wrong Address",
		path: "/",
	},
	{
		title: "How to Resume the Withdrawal Function",
		path: "/",
	},
];

export const faqArticles = [
	{
		title: "Development Environment Setup",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as execution environment. This guid will take you through the basic steps of setting up a development environment",
		path: "/",
	},
	{
		title: "Core-Tester provides set of instructions on how to run commands for creating and sending transactions?",
		description:
			"With our testnet up and running, the first thing we need to do is to connect to it. All interactions between ARK Nodes and the outside world happen through the Public API, which is a REST API facilitating different actions on the blockchain.",
		path: "/",
	},
	{
		title: "Setup Your Own Blockchain Explorer",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as execution environment. This guid will take you through the basic steps of setting up a development environment",
		path: "/",
	},
	{
		title: "Is it essential for you to understand and master the art of ARK Core Module development?",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as execution environment. This guid will take you through the basic steps of setting up a development environment",
		path: "/",
	},
];

export const article = {
	title: "Setup Your Own Blockchain Explorer",
	body: "jfdskfjdskajfkasjfasdf",
	category: "Portfolio",
	categoryIcon: "PortfolioTag",
	views: "32 908",
	image: "https://picsum.photos/seed/picsum/800/400",
	sections: [
		{
			title: "1. Introduction",
			id: "introduction",
			body:
				"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tempus imperdiet nulla malesuada pellentesque elit eget. Bibendum arcu vitae elementum curabitur. Eget lorem dolor sed viverra ipsum. Ut venenatis tellus in metus vulputate eu scelerisque. Mi bibendum neque egestas congue quisque egestas diam in. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nullam vehicula ipsum a arcu cursus. Sed libero enim sed faucibus. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Dui id ornare arcu odio ut sem nulla pharetra diam. Diam quam nulla porttitor massa.\n\n Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tempus imperdiet nulla malesuada pellentesque elit eget. Bibendum arcu vitae elementum curabitur. Eget lorem dolor sed viverra ipsum. Ut venenatis tellus in metus vulputate eu scelerisque. Mi bibendum neque egestas congue quisque egestas diam in. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nullam vehicula ipsum a arcu cursus. Sed libero enim sed faucibus. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Dui id ornare arcu odio ut sem nulla pharetra diam. Diam quam nulla porttitor massa.",
		},
		{
			title: "2. Create Account",
			id: "account",
			body:
				"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tempus imperdiet nulla malesuada pellentesque elit eget. Bibendum arcu vitae elementum curabitur. Eget lorem dolor sed viverra ipsum. Ut venenatis tellus in metus vulputate eu scelerisque. Mi bibendum neque egestas congue quisque egestas diam in. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nullam vehicula ipsum a arcu cursus. Sed libero enim sed faucibus. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Dui id ornare arcu odio ut sem nulla pharetra diam. Diam quam nulla porttitor massa.\n\n Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tempus imperdiet nulla malesuada pellentesque elit eget. Bibendum arcu vitae elementum curabitur. Eget lorem dolor sed viverra ipsum. Ut venenatis tellus in metus vulputate eu scelerisque. Mi bibendum neque egestas congue quisque egestas diam in. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nullam vehicula ipsum a arcu cursus. Sed libero enim sed faucibus. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Dui id ornare arcu odio ut sem nulla pharetra diam. Diam quam nulla porttitor massa.",
		},
	],
};
