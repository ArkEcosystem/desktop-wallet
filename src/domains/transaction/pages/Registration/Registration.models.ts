import { Contracts } from "@arkecosystem/platform-sdk";
import { ReadWriteWallet } from "@arkecosystem/platform-sdk-profiles";

export type RegistrationType = {
	label: string;
	value: string;
};

export type RegistrationForm = {
	transactionDetails: ({
		transaction,
		translations,
	}: {
		transaction: Contracts.SignedTransactionData;
		translations: any;
	}) => JSX.Element;

	signTransaction: ({ env, form, handleNext, profile, setTransaction, translations }: any) => Promise<void>;

	tabSteps: number;

	formFields: string[];

	component: ({
		activeTab,
		feeOptions,
		wallet,
	}: {
		activeTab: number;
		feeOptions: any;
		wallet: ReadWriteWallet;
	}) => JSX.Element;
};
