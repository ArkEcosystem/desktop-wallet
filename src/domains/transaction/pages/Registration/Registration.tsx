import { Address } from "app/components/Address";
import { Avatar } from "app/components/Avatar";
import { Button } from "app/components/Button";
import { Circle } from "app/components/Circle";
import { Form, FormField, FormLabel } from "app/components/Form";
import { Icon } from "app/components/Icon";
import { Input, InputPassword } from "app/components/Input";
import { Label } from "app/components/Label";
import { Page, Section } from "app/components/Layout";
import { Select } from "app/components/SelectDropdown";
import { useSelectionState } from "app/components/SelectionBar";
import { SelectNetwork } from "app/components/SelectNetwork";
import { StepIndicator } from "app/components/StepIndicator";
import { TabPanel, Tabs } from "app/components/Tabs";
import { TextArea } from "app/components/TextArea";
import { TransactionDetail } from "app/components/TransactionDetail";
import { ProfileFormField } from "domains/profile/components/ProfileFormField";
import { InputFee } from "domains/transaction/components/InputFee";
import { LedgerConfirmation } from "domains/transaction/components/LedgerConfirmation";
import { LinkCollection } from "domains/transaction/components/LinkCollection";
import { LinkList } from "domains/transaction/components/LinkList";
import { TotalAmountBox } from "domains/transaction/components/TotalAmountBox";
import { TransactionSuccessful } from "domains/transaction/components/TransactionSuccessful";
import React from "react";
import { useForm } from "react-hook-form";
import { styled } from "twin.macro";

type RegistrationProps = {
	addresses: any;
	formDefaultData: any;
	onDownload: any;
	networks: any;
	registrationTypes: any;
};

type Network = { name: string; label: string; value: string; icon: string; iconClass: string };
type RegistrationType = { label: string; value: string };

const FormWrapper = styled.div`
	.select-transparent {
		> div:first-child {
			width: 100%;
		}

		select {
			&:not([disabled]) {
				background-color: transparent;
				color: transparent;
			}
			option {
				color: var(--theme-color-neutral-900);
			}
		}
	}
`;

const RegistrationTypeDropdown = ({ className, register, registrationTypes }: any) => {
	return (
		<FormField data-testid="Registration__type" name="registrationType" className={`relative h-20 ${className}`}>
			<div className="mb-2">
				<FormLabel label="Registration Type" />
			</div>
			<div>
				<Select ref={register} data-testid="Registration__type-select" options={registrationTypes} />
			</div>
		</FormField>
	);
};

const getAddressInfo = (addresses: any[], address: string) => {
	return addresses.find((profile: any) => profile.address === address);
};

const getRegistrationByName = (registrationTypes: any[], registrationType: string) => {
	return registrationTypes.find((type: any) => type.value === registrationType);
};

const FirstStep = ({
	addresses,
	form,
	networks,
	registrationTypes,
}: {
	addresses: any;
	form: any;
	networks: Network[];
	registrationTypes: RegistrationType[];
}) => {
	const { register } = form;
	const { address, registrationType } = form.watch();

	return (
		<div data-testid="Registration__first-step">
			<h1 className="mb-0">Registration</h1>
			<div className="text-theme-neutral-dark">
				Select the type of registration and the address you want to register with.
			</div>

			<FormWrapper className="mt-8">
				<FormField name="network">
					<FormLabel>Network</FormLabel>
					<SelectNetwork networks={networks} />
				</FormField>

				<ProfileFormField
					formName="address"
					formLabel="Address"
					profiles={addresses}
					selectedProfile={getAddressInfo(addresses, address)}
					register={register}
					withoutAdditional={true}
					className="mt-8"
				/>

				<RegistrationTypeDropdown
					selectedType={getRegistrationByName(registrationTypes, registrationType)}
					registrationTypes={registrationTypes}
					className="mt-8"
					register={register}
				/>
			</FormWrapper>
		</div>
	);
};

const SecondStep = ({ form }: { form: any }) => {
	const { register } = form;
	const selectionBarState = useSelectionState(1);

	return (
		<div data-testid="Registration__second-step">
			<h1 className="mb-0">Register Business</h1>
			<div className="text-theme-neutral-dark">
				Select the type of registration and the address you want to register with.
			</div>

			<div>
				<TransactionDetail border={false} className="pb-8">
					<FormField name="name" className="font-normal">
						<FormLabel required>Name</FormLabel>
						<Input type="text" ref={register} />
					</FormField>

					<FormField name="description" className="mt-8 font-normal">
						<FormLabel required>Description</FormLabel>
						<TextArea ref={register} />
					</FormField>

					<FormField name="website" className="mt-8 font-normal">
						<FormLabel>Website</FormLabel>
						<Input type="website" ref={register} />
					</FormField>
				</TransactionDetail>

				<TransactionDetail className="pb-8">
					<LinkCollection
						title="Repository"
						description="Show your projects through your repository"
						types={[
							{ label: "BitBucket", value: "bitbucket" },
							{ label: "GitHub", value: "github" },
							{ label: "GitLab", value: "gitlab" },
						]}
						typeName="repository"
					/>
				</TransactionDetail>

				<TransactionDetail className="pb-8">
					<LinkCollection
						title="Social Media"
						description="Tell people more about yourself through social media"
						types={[
							{ label: "Facebook", value: "facebook" },
							{ label: "Twitter", value: "twitter" },
							{ label: "LinkedIn", value: "linkedin" },
						]}
						typeName="media"
					/>
				</TransactionDetail>

				<TransactionDetail className="pb-8">
					<LinkCollection
						title="Photo and Video"
						description="Get more users and add more information about yourself"
						types={[
							{ label: "YouTube", value: "youtube" },
							{ label: "Vimeo", value: "vimeo" },
							{ label: "Flickr", value: "flickr" },
						]}
						typeName="files"
						selectionTypes={["flickr"]}
						selectionTypeTitle="Avatar"
					/>
				</TransactionDetail>

				<TransactionDetail className="pt-6 pb-0">
					<FormField name="fee">
						<FormLabel>Fee ARK</FormLabel>

						<InputFee selectionBarState={selectionBarState} defaultValue={25} min={1} max={100} step={1} />
					</FormField>
				</TransactionDetail>
			</div>
		</div>
	);
};

const ThirdStep = () => {
	const links = [
		{
			link: "http://github.com/robank",
			type: "github",
		},
		{
			link: "http://gitlab.com/robank",
			type: "gitlab",
		},
		{
			link: "http://bitbucket.com/robank",
			type: "bitbucket",
		},
		{
			link: "http://npmjs.com/robank",
			type: "npm",
		},
	];

	return (
		<div data-testid="Registration__third-step">
			<div>
				<h1 className="mb-0">Transaction Review</h1>
				<p className="text-theme-neutral-dark">Check the information again before voting</p>
			</div>
			<div className="mt-4">
				<TransactionDetail
					border={false}
					label="Network"
					extra={
						<div className="ml-1 text-theme-danger">
							<Circle className="bg-theme-background border-theme-danger-light" size="lg">
								<Icon name="Ark" width={20} height={20} />
							</Circle>
						</div>
					}
				>
					<div className="flex-auto font-semibold truncate text-theme-neutral-800 max-w-24">
						ARK Ecosystem
					</div>
				</TransactionDetail>

				<TransactionDetail
					extra={<Avatar size="lg" address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK" />}
					className="pt-4"
				>
					<div className="mb-2 text-sm font-semibold text-theme-neutral">
						<span className="mr-1">Sender</span>
						<Label color="warning">
							<span className="text-sm">Your address</span>
						</Label>
					</div>
					<Address address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK" walletName={"ROBank"} />
				</TransactionDetail>

				<TransactionDetail
					label="Type"
					extra={
						<div>
							<Circle className="border-black bg-theme-background" size="lg">
								<Icon name="Business" width={20} height={20} />
							</Circle>
						</div>
					}
				>
					Business Registration
				</TransactionDetail>

				<TransactionDetail label="Name">ROBank Eco</TransactionDetail>

				<TransactionDetail label="Description">Not a trustworthy bank</TransactionDetail>

				<TransactionDetail label="Website">
					<a href="http://robank.com" target="_blank" rel="noopener noreferrer" className="link">
						http://robank.com
					</a>
				</TransactionDetail>

				<TransactionDetail className="mb-2">
					<LinkList
						title="Repository"
						description="Show your projects through the repository"
						links={links}
					/>
				</TransactionDetail>

				<div>
					<TotalAmountBox transactionAmount="0.00" transactionFee="0.09660435" />
				</div>
			</div>
		</div>
	);
};

const FourthStep = ({ form, passwordType }: { form: any; passwordType: "mnemonic" | "password" | "ledger" }) => {
	const { register } = form;

	return (
		<div data-testid="Registration__fourth-step">
			{passwordType !== "ledger" && (
				<div>
					<h1 className="mb-0">Authenticate</h1>
					<div className="text-theme-neutral-dark">
						Enter your twelve word mnemonic to authenticate the transaction.
					</div>

					<div className="mt-8">
						<FormField name="name">
							<FormLabel>{passwordType === "mnemonic" ? "Mnemonic" : "Encryption Password"}</FormLabel>
							<InputPassword name={passwordType} ref={register} />
						</FormField>

						<FormField name="name" className="mt-8">
							<FormLabel>2nd Mnemonic</FormLabel>
							<InputPassword name="secondMnemonic" ref={register} />
						</FormField>
					</div>
				</div>
			)}

			{passwordType === "ledger" && (
				<div>
					<h1>Confirm Your Transaction</h1>
					<LedgerConfirmation />
				</div>
			)}
		</div>
	);
};

export const FifthStep = () => (
	<TransactionSuccessful>
		<TransactionDetail
			label="Transaction Type"
			extra={
				<Circle className="border-black" size="lg">
					<Icon name="Business" width={20} height={20} />
				</Circle>
			}
		>
			Business Registration
		</TransactionDetail>
		<TransactionDetail label="Name">ROBank Eco</TransactionDetail>
		<TransactionDetail label="Description">Not a trustworthy bank</TransactionDetail>
		<TransactionDetail label="Website">
			<a href="http://robank.com" target="_blank" rel="noopener noreferrer" className="link">
				http://robank.com
			</a>
		</TransactionDetail>
		<TransactionDetail
			label="Amount"
			className="pb-0"
			extra={
				<div className="ml-1 text-theme-danger">
					<Circle className="bg-theme-background border-theme-danger-light" size="lg">
						<Icon name="Sent" width={22} height={22} />
					</Circle>
				</div>
			}
		>
			1.09660435 ARK
		</TransactionDetail>
	</TransactionSuccessful>
);

export const Registration = ({
	addresses,
	formDefaultData,
	onDownload,
	networks,
	registrationTypes,
}: RegistrationProps) => {
	const form = useForm({ mode: "onChange", defaultValues: formDefaultData });
	const [activeTab, setActiveTab] = React.useState(1);
	const { formState } = form;
	const { isValid } = formState;

	const handleBack = () => {
		setActiveTab(activeTab - 1);
	};

	const handleNext = () => {
		setActiveTab(activeTab + 1);
	};

	const crumbs = [
		{
			route: "portfolio",
			label: "Go back to Portfolio",
		},
	];

	return (
		<Page crumbs={crumbs}>
			<Section className="flex-1">
				<Form className="max-w-xl mx-auto" context={form} onSubmit={(data: any) => onDownload(data)}>
					<Tabs activeId={activeTab}>
						<StepIndicator size={7} activeIndex={activeTab} />

						<div className="mt-8">
							<TabPanel tabId={1}>
								<FirstStep
									addresses={addresses}
									form={form}
									networks={networks}
									registrationTypes={registrationTypes}
								/>
							</TabPanel>
							<TabPanel tabId={2}>
								<SecondStep form={form} />
							</TabPanel>
							<TabPanel tabId={3}>
								<ThirdStep />
							</TabPanel>
							<TabPanel tabId={4}>
								<FourthStep form={form} passwordType="mnemonic" />
							</TabPanel>
							<TabPanel tabId={5}>
								<FourthStep form={form} passwordType="password" />
							</TabPanel>
							<TabPanel tabId={6}>
								<FourthStep form={form} passwordType="ledger" />
							</TabPanel>
							<TabPanel tabId={7}>
								<FifthStep />
							</TabPanel>

							<div className="flex justify-end mt-8 space-x-3">
								{activeTab < 7 && (
									<Button
										disabled={activeTab === 1}
										data-testid="Registration__back-button"
										variant="plain"
										onClick={handleBack}
									>
										Back
									</Button>
								)}

								{activeTab < 4 && (
									<Button
										data-testid="Registration__continue-button"
										disabled={!isValid}
										onClick={handleNext}
									>
										Continue
									</Button>
								)}

								{activeTab >= 4 && activeTab < 7 && (
									<Button
										data-testid="Registration__send-button"
										disabled={!isValid}
										onClick={handleNext}
									>
										<Icon name="Send" className="mr-2" width={20} height={20} />
										Send
									</Button>
								)}

								{activeTab === 7 && (
									<div className="flex justify-end space-x-3">
										<Button data-testid="Registration__wallet-button" variant="plain">
											Back to wallet
										</Button>

										<Button
											type="submit"
											data-testid="Registration__download-button"
											variant="plain"
										>
											<Icon name="Download" className="mr-2" />
											Download
										</Button>
									</div>
								)}
							</div>
						</div>
					</Tabs>
				</Form>
			</Section>
		</Page>
	);
};
