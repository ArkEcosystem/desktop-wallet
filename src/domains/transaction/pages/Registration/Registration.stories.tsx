import React from "react";

import { Registration } from "./Registration";

export default { title: "Domains / Transaction / Pages / Registration" };

const defaultFormValues = {
	networks: [
		{
			icon: "Ark",
			name: "Ark Ecosystem",
			className: "text-theme-danger-400 border-theme-danger-200",
		},
		{
			icon: "Bitcoin",
			name: "Bitcoin",
			className: "text-theme-warning-400 border-theme-warning-200",
		},
		{
			icon: "Ethereum",
			name: "Ethereum",
			className: "text-theme-neutral-800 border-theme-neutral-600",
		},
	],
	registrationTypes: [
		{
			value: "business",
			label: "Business",
		},
	],
	formDefaultData: {
		network: null,
		address: null,
	},
	addresses: [
		{
			address: "FJKDSALJFKASLJFKSDAJFKFKDSAJFKSAJFKLASJKDFJ",
			walletName: "My Wallet",
			avatarId: "FJKDSALJFKASLJFKSDAJFKFKDSAJFKSAJFKLASJKDFJ",
			formatted: "My Wallet FJKDSALJFKASL...SAJFKLASJKDFJ",
		},
	],
};

export const Default = () => <Registration {...defaultFormValues} onDownload={() => alert("download")} />;
