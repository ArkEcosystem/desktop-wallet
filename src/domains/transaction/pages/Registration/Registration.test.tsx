/* eslint-disable @typescript-eslint/require-await */

import { availableNetworksMock } from "domains/network/data";
import { createMemoryHistory } from "history";
import React from "react";
import { Route } from "react-router-dom";
import { act, fireEvent, renderWithRouter, waitFor } from "testing-library";
import { identity } from "tests/fixtures/identity";

import { Registration } from "./Registration";

let rendered: RenderResult;
let defaultFormValues = {};

describe("Registration", () => {
	beforeEach(() => {
		const history = createMemoryHistory();
		const registrationURL = `/profiles/${identity.profiles.bob.id}/transactions/registration`;

		history.push(registrationURL);

		defaultFormValues = {
			networks: availableNetworksMock,
			registrationTypes: [
				{
					value: "business",
					label: "Business",
				},
			],
			formDefaultData: {
				network: null,
				address: null,
			},
			addresses: [
				{
					address: "FJKDSALJFKASLJFKSDAJFKFKDSAJFKSAJFKLASJKDFJ",
					walletName: "My Wallet",
					avatarId: "FJKDSALJFKASLJFKSDAJFKFKDSAJFKSAJFKLASJKDFJ",
					formatted: "My Wallet FJKDSALJFKASL...SAJFKLASJKDFJ",
				},
			],
			onDownload: jest.fn(),
		};

		rendered = renderWithRouter(
			<Route path="/profiles/:profileId/transactions/registration">
				<Registration {...defaultFormValues} />
			</Route>,
			{
				routes: [registrationURL],
				history,
			},
		);
	});

	it("should render 1st step", async () => {
		const { asFragment, getByTestId } = rendered;

		expect(getByTestId("Registration__first-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});

	it("should should go back", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__back-button"));
		});

		expect(getByTestId("Registration__first-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});

	it("should render 2nd step", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});

		expect(getByTestId("Registration__second-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});

	it("should render 3rd step", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});

		expect(getByTestId("Registration__third-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});

	it("should render 4th step", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});

		expect(getByTestId("Registration__fourth-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});

	it("should render 5th step", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__send-button"));
		});

		expect(getByTestId("TransactionSuccessful")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});

	it("should submit", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("Registration__download-button"));
		});

		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(1);
		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});

	it("should select registration type", async () => {
		const { asFragment, getByTestId } = rendered;

		const toggle = getByTestId("select-list__toggle-button");

		act(() => {
			fireEvent.click(toggle);
		});

		const firstOption = getByTestId("select-list__toggle-option-0");
		expect(firstOption).toBeTruthy();

		act(() => {
			fireEvent.click(firstOption);
		});

		expect(getByTestId("select-list__input")).toHaveValue("business");
		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});
});
