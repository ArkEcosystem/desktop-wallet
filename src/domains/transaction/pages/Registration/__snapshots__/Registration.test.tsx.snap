// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration should render 1st step 1`] = `[Function]`;

exports[`Registration should render 2nd step 1`] = `[Function]`;

exports[`Registration should render 3rd step 1`] = `[Function]`;

exports[`Registration should render 4th step 1`] = `[Function]`;

exports[`Registration should render 5th step 1`] = `[Function]`;

exports[`Registration should select registration type 1`] = `[Function]`;

exports[`Registration should should go back 1`] = `[Function]`;

exports[`Registration should submit 1`] = `[Function]`;
